### ISSUE TITLE
Changed behavior of the Vim colorscheme

### ISSUE THREAD

**Issue by IngoMeyer441** [neovim/neovim#26855]
### Problem

Since commit 720a3518e3d2eee745bee1d856c92e7c1d84549f (PR #26658) the legacy Vim colorscheme behaves slightly different. Before the PR the commands

```vim
colorscheme vim
hi NormalFloat
```

prints

```
NormalFloat    xxx guibg=NvimDarkGrey1
                   links to Pmenu
```

but since the PR, it outputs

```
NormalFloat    xxx links to Pmenu
```

This affects legacy colorschemes which rely on the old Vim colorscheme.

### Steps to reproduce

nvim --clean
:colorscheme vim
:hi NormalFloat

### Expected behavior

Links between highlighting groups should be preserved to stay compatible with the legacy Vim colorscheme.

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1914+g720a3518e

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Manjaro Linux 23.1.2 Vulcan

### Terminal name/version

kitty 0.31.0

### $TERM environment variable

tmux-256color

### Installation

build from repo

**Comment by clason** [comment]
Links _are_ preserved (as they take precedence over defined attributes), so there should be no observable(!) difference. What is the actual impact on a legacy colorscheme? Please give screenshots.

**Comment by IngoMeyer441** [comment]
> Links are preserved

Sorry, I was too stupid to read the output of `:highlight` correctly, the only difference is the dropped guibg color.

This screenshot uses [a fork of the vim-one colorscheme](https://github.com/IngoMeyer441/vim-one) before commit 720a3518e3d2eee745bee1d856c92e7c1d84549f:

![before](https://github.com/neovim/neovim/assets/1449087/61d6d1f5-8e1f-46d2-b7fb-78ae9d028562)

This is the floating window color after commit 720a3518e3d2eee745bee1d856c92e7c1d84549f:

![after](https://github.com/neovim/neovim/assets/1449087/10bff3c2-98e2-4e3e-88de-d2ba54078f4f)

But I already found the reason... In my fork, I sourced the `colors/vim.vim` script to make the legacy vim-one colorscheme compatible with Neovim 0.10. But it should of course be `colors/vim.lua` now. So this issue can be closed, sorry for stealing your time.
